### ISSUE TITLE
ScriptHost stopped after WebHostRpcWorkerChannelManager stopped 

### ISSUE THREAD

**Issue by mhoeger** [Azure/azure-functions-host#5785]
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L130

This can lead to a "Did not find initialized language worker" failure of a function execution for languages with webhost level workers. 

Order of events:
- Execution begins for function X
- StopApplication() from somewhere
- Web host level channels stopped [here](https://github.com/Azure/azure-functions-host/blob/f987062b51d36b2a66f8bb2f1aeae9a4012d48ed/src/WebJobs.Script/Workers/Rpc/RpcInitializationService.cs#L79)
- Execution continues... does not find initialized language workers
- ScriptHost stopped [here](https://github.com/Azure/azure-functions-host/blob/f987062b51d36b2a66f8bb2f1aeae9a4012d48ed/src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs#L379) which would prevent the "did not find initialized language workers"

Applicable to java always. Applicable to python always in production. Applicable to node in production (with placeholders)

**Comment by mhoeger** [comment]
Actually @brettsam - I'll keep you on this issue if that's alright! This is what I've seen happen and believe to be the cause. Does that make sense to you too? Basically web host level worker channels are being shut down before jobhost knows it's shutting down

**Comment by brettsam** [comment]
Yes that that seems right. Here's where those things are registered -- https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L129-L136. 

I seem to recall that @fabiocav, @yojagad, @pragnagopa and I talked about this HostedService shutdown ordering a while back and how it was subtly different for WebHost vs ScriptHost (due to .NET Core behavior). I believe that this HostedServiceManager was added to help combat that -- but maybe it's still biting you here.

**Comment by pragnagopa** [comment]
@yojagad should have most context here. JobHost should always be stopped before  webhost scope level: RpcInitializationService is shutdown. WebHost level worker channels should shutdown after jobhost gracefully shuts down.

**Comment by yojagad** [comment]
Maybe moving WebHostLanguageWorkerChannel shutdown to OuterStopAsync() may achieve the behavior we want here..
